### ISSUE TITLE
ClientRequest.prototype.setHeader don't handle Array values

### ISSUE THREAD

**Issue by ronny-springer** [jhiesey/stream-http#73]
Hi, I want to set 'access-control-allow-origin' header property via node-fetch (version: "1.6.3"):
```
const fetch = require('node-fetch');
promise = fetch(url, {headers: { 
		'Access-Control-Allow-Origin': 'http://google.com',
		}})
		.catch(error => console.log(error));
```
But the value was set to property name. So, I decide to debug it. It's inside the request.js (lib folder).
I'm coming from this code line:
```
Object.keys(opts.headers).forEach(function (name) {
	self.setHeader(name, opts.headers[name])
})
```
 I'm passing the name: "access-control-allow-origin" and the value: Array[1] that contains "http://google.com"
```
ClientRequest.prototype.setHeader = function (name, value) {
	...
	self._headers[lowerName] = {
		name: name,
		value: value // <- The value of the array is not handled
	}
}
```
@jhiesey What's the best level to fix it?

**Comment by jhiesey** [comment]
This is a bug in stream-http. This module should support arrays as header values the same way as node. I will try to get this fixed this week.

**Comment by jhiesey** [comment]
Fix released in v2.7.0
